composer.json

```diff
@@ -42,9 +46,6 @@
     ]
   },
   "config": {
-    "sort-packages": true,
-    "platform": {
-      "php": "7.3"
```

Comment:
Removing this will ensure that composer is never successfully installed on a machine that doesn't have PHP 7.2+ and to be technical it's actually PHP 7.2.5 that is required to run Composer locally. However, I added this so composer can be executed on the Docker image and ignore the requirement for my local box to get setup.

Reply:
It's now moved to the `requires` section:

https://github.com/xwp/block-scaffolding-wp/blob/83aa3ee9dbb20282fc1b3f9285e43c7b6ea42393/composer.json#L9

That should trigger an error if the PHP version doesn't match, right?

Reply:
I don't believe there is actually a requirement to run PHP 7.2 for the plugin code, unless it's for the PSR-4 autoloader — which I doubt. Otherwise, there are only dependencies for development and not for the built plugin.

Reply:
Yeah, that's a good point! Only the development dependencies really need PHP 7.2 or later.

The issue with setting the platform PHP version to 7.2 is that all dependencies (including the non-dev) in the lock file will be resolved as if PHP 7.2 was present. So using `require` or `platform` still doesn't solve the issue.

I'll create an issue to address this.

Reply:
What I'm saying is I don't want to trigger an error for development dependencies. The required PHP version will be available to the Docker container. If we force the local box to have PHP 7.2+ then I and many others who have not updated PHP locally, because I never use it and always run containers, will be stuck.

Reply:
> What I'm saying is I don't want to trigger an error for development dependencies. The required PHP version will be available to the Docker container.

Could this be resolved by running all commands inside the container -- even `npm run docker -- npm install`?

Reply:
I believe you have a chicken egg situation, as there are dependencies to run `npm install` and cannot `vagrant up` until you've done so.

Reply:
With Docker on the host one could do `npm run docker -- npm install` right away (since it uses `docker-compose run` instead of `exec`) and with Vagrant they would need to start Vagrant first, yes.

Reply:
Correct, but it's not very clear you need to do that and that also means Docker becomes a legitimate development dependency, which kind of makes the whole vagrant wrapper not needed as you can avoid port collisions and edit the host file directly and support SSL all from Docker. 🤷‍♂

Reply:
> Correct, but it's not very clear you need to do that and that also means Docker becomes a legitimate development dependency.

The requirements are listed out in the "Requirements" section of the readme so they seem to be pretty well defined. One thing we should make clear is that Node.js and Composer is not required for using the plugin but only the development.

We require that developers:

1. Have PHP 7.2+, Composer and Node.js installed locally (per requirements section in the readme) to run the basic dev tooling such as linters and tests (both PHP and JS).

or

2. Use the bundled development environment which has all the tooling included.

So if a developer doesn't have the required dependencies installed locally, they _must_ use the virtual environment. It would be almost impossible to make this work if users had only Node _or_ PHP available locally.